src/components/DisplayMenuAndOrder.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { Component } from "react";
+import { getData } from "../modules/productData";
+import axios from "axios";
+import { authenticate } from "../modules/authenticate";
+
+class DisplayMenuAndOrder extends Component {
+  state = {
+    productData: [],
+    message: {},
+    orderDetails: {},
+    showOrder: false
+  };
+
+  componentDidMount() {
+    this.getProductData();
+  }
+
+  async getProductData() {
+    let result = await getData();
+    this.setState({ productData: result.data.products });
+  }
+
+  async addToOrder(event) {
+    let id = event.target.parentElement.dataset.id;
+    let result;
+    if (this.state.orderDetails.hasOwnProperty("id")) {
+      result = await axios.put(
+        `http://localhost:3000/api/orders/${this.state.orderDetails.id}`,
+        { product_id: id }
+      );
+    } else {
+      result = await axios.post("http://localhost:3000/api/orders", {
+        product_id: id
+      });
+    }
+    this.setState({
+      message: { id: id, message: result.data.message },
+      orderDetails: result.data.order
+    });
+  }
+
+  finalizeOrder() {
+    this.setState({
```

Comment:
How do you plan to communicate this to the backend? What attribute should it be updating?

Reply:
After a discussion, we would initially say that this is a gap in our current configuration where we would want to be able to pull back a confirmed order as a GET request. And that the confirm button could, in theory, serve to trigger a GET request that would confirm to the restaurant that the user has confirmed the order. Based on our understanding the order is saved into the data base with POST and PUT requests, so there would be no need for there to be an extra attribute in the order table necessarily if we could utilise a GET command (this would require an update on the API routes).

We are slightly confused if, in your eyes, this is an immediate concern for the weekend challenge that requires an update. Or if your questions were more in theory about what we would need to next implement or think about.

Reply:
Well, it is definitely part of the ordering process and I think you should aim to fix this both here in the client and on the API-side.

Reply:
We will do a new branch and PR for this update